gcf_data_mapper/parsers/document.py

```diff
@@ -1,19 +1,152 @@
+from enum import Enum
 from typing import Any, Optional
+from urllib.parse import urlparse
 
 import click
 import pandas as pd
 
 
+class DocumentColumns(Enum):
+    SOURCE_URL = "Document page permalink"
+    TITLE = "Title"
+    TRANSLATED_FILES = "Translated files"
+    TRANSLATED_TITLES = "Translated titles"
+    TYPE = "Type"
+
+
+def contains_duplicate_urls(urls: list[str]) -> bool:
+    """
+    Checks a list of urls for any duplicate entries
+
+    param: list[str] urls: A list of urls
+    return bool: Returns true if duplicate urls are present
+    """
+
+    # Convert all URLs to lowercase for case-insensitive comparison
+    lowercase_urls = [url.lower() for url in urls]
+    return len(lowercase_urls) != len(set(lowercase_urls))
+
+
+def contains_empty_urls(urls: list[str]) -> bool:
+    """
+    Checks a list of urls for any empty entries
+
+    param: list[str] urls: a list of urls
+    return bool: Returns true if empty urls are present, or false if not
+    """
+    for url in urls:
+        if not url.strip():
+            return True
+    return False
+
+
+def contains_invalid_paths(urls: list[str]) -> bool:
+    """
+    Checks a list of urls for any malformed entries
+
+    param: list[str] urls: A list of urls
+    return bool: Returns true if malformed urls are present, or false if not
+    """
+    for url in urls:
+        parsed_url = urlparse(url)
+        path = parsed_url.path
+        if any(not (c.isalnum() or c in "/-_.") for c in path):
```

Comment:
Does this work for URLs with a colon (:) in it?

Reply:
nope, but shall update